WordPress/gutenberg
# [RNMobile] Fix issue related to receiving undefined ref in text color format

## What?


In rare cases, probably related to slow performance in low-end devices, the ref object `contentRef` passed from the `RichText` component to the `TextColorEdit` component can be undefined upon mounting. This ref object is used to get the background color of the text and use it in the toolbar button. However, the logic of the text color format assumed that this ref is always defined, leading to a crash.

## Why?

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/6189.
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/6126.

## How?


This PR addresses this issue by checking the existence of the ref object `contentRef`.

## Testing Instructions




**NOTE:** I couldn't find a reliable way to reproduce the issue except modifying the code.

1. Apply the following patch:
```patch
diff --git forkSrcPrefix/packages/block-editor/src/components/rich-text/native/index.native.js forkDstPrefix/packages/block-editor/src/components/rich-text/native/index.native.js
index ab465b2441154920994a6643a529bd6e237c1900..1710c2d5423b4c5f8c8a7b1c8928f2031f14cebc 100644
--- forkSrcPrefix/packages/block-editor/src/components/rich-text/native/index.native.js
+++ forkDstPrefix/packages/block-editor/src/components/rich-text/native/index.native.js
@@ -1334,7 +1334,7 @@ export class RichText extends Component {
 				{ isSelected && (
 					<>
 						<FormatEdit
-							forwardedRef={ this._editor }
+							forwardedRef={ undefined }
 							formatTypes={ formatTypes }
 							value={ record }
 							onChange={ this.onFormatChange }
```
2. Open the app.
3. Create/open a post.
4. Select the first Paragraph block.
5. Type some text and select it.
6. Tap on the 💧 format button located in the bottom toolbar.
7. Select a color.
8. Observe that no crash/exception is produced.
9. Observe that the selected color has been applied to the text.

### (UPDATE) Reload React Native
I noticed while testing another way to reproduce it, although it only works when connecting the app with a local Metro server. I don't think this behavior can be reproduced in a production version, however, it confirms the fact that `contentRef` can be undefined in some cases.

1. Open the app.
2. Create/open a post.
3. Select the first Paragraph block.
4. Type some text and select it.
5. Tap on the 💧 format button located in the bottom toolbar.
6. Select a color.
7. Dismiss the bottom sheet.
8. Select the text.
9. Reload React Native via the debug menu/metro server.
10. Observe that no crash/exception is produced.

### Testing Instructions for Keyboard

N/A

## Screenshots or screencast 

**NOTE:** The theme has the background color set to `#ABB8C3`.

| Common case: `contentRef` is defined  | `contentRef` is undefined |
|--------|--------|
| <img src=https://github.com/WordPress/gutenberg/assets/14905380/a32db4f0-1be3-4874-ac9a-aa2543959d06 width=300> | <img src=https://github.com/WordPress/gutenberg/assets/14905380/19e9ffd4-fc54-4804-94fa-1f2b1feeab68 width=300> |
